src/sdk/registrar.ts

```diff
@@ -47,23 +47,40 @@ export class FuelnameRegistrar {
     }
   }
 
-  async mintDomain(domain: string, years: number, txParams: TxParams, price: DomainPrice): Promise<CallResult<string>> {
+  private async getResourcesToSpend(quantities: CoinQuantityLike[]) {
+    if (this.accountOrProvider instanceof Account) {
+      return await this.accountOrProvider.getResourcesToSpend(quantities);
+    }
+
+    throw new Error('Account is not available');
+  }
+
+  async mintDomain(domain: string, years: number, txParams: TxParams, price: DomainPrice): Promise<ScriptTransactionRequest> {
     try {
       const subdomain = removeFuelDomainSuffix(domain);
 
-      const scopeCall = await this.writeRegistrarContract!
+      const request = await this.writeRegistrarContract!
         .functions
         .mint_domain(subdomain, years)
-        .callParams({forward: {amount: price.value, assetId: price.assetId}})
```

Comment:
We should probably keep the callParams, otherwise the contract won't see these tokens transferred